src/elona/god.cpp

```diff
@@ -478,6 +235,159 @@ void switch_religion()
 }
 
 
+static optional<TurnResult> _create_faith_reward_servant()
+{
+    bool no_more_servants = false;
+    int servant_count = 0;
+
+    for (int cnt = 1; cnt < 16; ++cnt)
+    {
+        if (cdata[cnt].state() != Character::State::empty)
+        {
+            if (cdatan(2, cnt) == u8"core.servant"s)
+            {
+                ++servant_count;
+                if (servant_count >= 2)
+                {
+                    no_more_servants = true;
+                    txt(i18n::s.get("core.locale.god.pray.servant.no_more"));
+                    break;
+                }
+            }
+        }
+    }
+    if (!no_more_servants)
+    {
+        if (chara_get_free_slot_ally() == 0)
+        {
+            no_more_servants = true;
+            txt(i18n::s.get("core.locale.god.pray.servant.party_is_full"));
+        }
+    }
+    if (no_more_servants)
+    {
+        txt(i18n::s.get("core.locale.god.pray.servant.prompt_decline"));
+        if (yes_no())
+        {
+            ++game_data.god_rank;
+        }
+        return TurnResult::turn_end;
+    }
+
+    auto god_id = cdata[cc].god_id;
+    auto god_data = the_god_db[god_id];
+    assert(god_data);
+
+    if (!god_data)
+    {
+        txt("Cannot find god ID '"s + god_id + "'."s,
+            Message::color(ColorIndex::red));
+        return TurnResult::turn_end;
+    }
+
+    auto chara = the_character_db[god_data->servant_id];
+    if (!chara)
+    {
+        txt("Cannot find character ID '"s + god_data->servant_id.get() + "'."s,
+            Message::color(ColorIndex::red));
+        return TurnResult::turn_end;
+    }
+
+    txt(i18n::s.get_m("locale.god", god_id, "servant"),
+        Message::color{ColorIndex::blue});
+
+    flt();
+    dbid = chara->legacy_id;
+
+    novoidlv = 1;
+    chara_create(56, dbid, -3, 0);
+    rc = 56;
+    new_ally_joins();
+
+    return none;
+}
+
+static optional<TurnResult> _create_faith_reward_gemstone()
+{
+    auto god_id = cdata[cc].god_id;
+    auto god_data = the_god_db[god_id];
+    assert(god_data);
+
+    if (!god_data)
+    {
+        lua::print_data_error("god", god_id);
+        return TurnResult::turn_end;
+    }
+
+    for (size_t i = 1; i <= god_data->items.size(); i++)
+    {
+        sol::object item_proto_obj = god_data->items[i];
+        if (item_proto_obj != sol::lua_nil && item_proto_obj.is<sol::table>())
+        {
+            lua::ConfigTable item_proto = item_proto_obj.as<sol::table>();
+
+            auto only_once = item_proto.optional_or<bool>("_only_once", false);
+
+            if (only_once)
+            {
+                auto id = item_proto.optional<std::string>("id");
+                auto item = the_item_db[*id];
+                if (item)
+                {
+                    if (only_once && itemmemory(1, item->legacy_id))
+                    {
+                        continue;
```

Comment:
In this case, Mani, Kumi, Jure, and Lulwy give a potion of cure corruption instead (In other words, gods who give gem stone, not secret treasure of XXX, give a potion of cure corruption).